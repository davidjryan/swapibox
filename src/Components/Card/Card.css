.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: 'Poller One', serif;
  font-weight: bold;
  background: #585858;
  width: 350px;
  margin: 5px;
  border-radius: 5px; }

hr {
  margin: 5px; }

.card:hover {
  box-shadow: 0 8px 16px 0 #734F51; }

.card-info-title {
  font-weight: bold;
  font-size: 1.4em;
  margin-bottom: 0; }

.card-title {
  border-bottom: 2px solid #191A21;
  display: block;
  width: 100%;
  margin-bottom: 0px;
  padding-bottom: 15px; }
