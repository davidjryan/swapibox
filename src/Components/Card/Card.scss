@import "../../variables.scss";

.card {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: 'Poller One', serif;
  font-weight: bold;
  background: $body-medium;
  width: 350px;
  margin: 5px;
  border-radius: 5px;
}
hr {
  margin: 5px;
}

.card-1:hover {
  box-shadow: 0 0 5px 5px $lightsaber-green;
}

.card-2:hover {
  box-shadow: 0 0 5px 5px $lightsaber-blue;
}

.card-3:hover {
  box-shadow: 0 0 5px 5px $lightsaber-red;
}

.card-info-title {
  font-weight: bold;
  font-size: 1.4em;
  margin-bottom: 0;
}

.card-title {
  border-bottom: 2px solid $text-dark;
  display: block;
  width: 100%;
  margin-bottom: 0px;
  padding-bottom: 15px;
}
