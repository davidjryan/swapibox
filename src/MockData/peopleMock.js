module.exports = [

    {
      Name: "Luke Skywalker",
      Species: "Human",
      Language: "Galactic Basic",
      "Homeworld Population": "200000",
      Homeworld: "Tatooine",
      favorite: false
    },
    {
      Name: "C-3PO",
      Species: "Droid",
      Language: "n/a",
      "Homeworld Population": "200000",
      Homeworld: "Tatooine",
      favorite: false
    },
    {
      Name: "R2-D2",
      Species: "Droid",
      Language: "n/a",
      "Homeworld Population": "4500000000",
      Homeworld: "Naboo",
      favorite: false
    }

];
