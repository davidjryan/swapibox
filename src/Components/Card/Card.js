import React from 'react';
import Button from '../Button/Button';
import './Card.css';
import PropTypes from 'prop-types';

const Card = ({ cardInformation, toggleFav }) => {
  const cardKeys = Object.keys(cardInformation);

  const cardDetails = cardKeys.map( (discreteInfo) => {
    if (discreteInfo !== 'Name' && discreteInfo !== 'favorite') {
      return (
        <div key={`${discreteInfo} - ${cardInformation.Name}`}>
          {/* make this div and it's contents into a component
          so that we can move the card rendering into this and
          apply unique classes to each card as it appends */}

          <p className="card-info-title">{discreteInfo}</p>
          <p className="card-info-body">{cardInformation[discreteInfo]}</p>
        </div>
      );
    }
    return null;
  });

  return (
    <div className="card">
      <h2 className="card-title ">
        {cardInformation.Name}
      </h2>
      <Button
        givenClass={'favorite-button'}
        text="" click={toggleFav}
        category={cardInformation}
        cardInformation={cardInformation}/>
      { cardDetails }
    </div>
  );
};

Card.propTypes = {
  cardInformation: PropTypes.object.isRequired,
  toggleFav: PropTypes.func.isRequired
};

export default Card;
