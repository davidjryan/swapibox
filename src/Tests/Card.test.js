import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from '../Components/Card/Card';
import peopleMock from '../MockData/peopleMock';

describe('Card', () => {
  let wrapper;

  it('should create an instance of a card', () => {
    const mockFunc = jest.fn();
    wrapper = shallow(<Card
      cardInformation={peopleMock[0]}
      toggleFav={mockFunc}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should have a button that can be clicked', () => {
    const mockFunc = jest.fn();
    const wrapper = mount(<Card
      cardInformation={peopleMock[0]}
      toggleFav={mockFunc}
    />);
    const button = wrapper.find('button');

    expect(mockFunc).toHaveBeenCalledTimes(0);
    button.simulate('click');
    expect(mockFunc).toHaveBeenCalledTimes(1);
  });
});
