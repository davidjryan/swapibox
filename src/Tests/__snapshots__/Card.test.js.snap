// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should create an instance of a card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    cardInformation={
        Object {
            "Homeworld": "Tatooine",
            "Homeworld Population": "200000",
            "Language": "Galactic Basic",
            "Name": "Luke Skywalker",
            "Species": "Human",
            "favorite": false,
          }
    }
    toggleFav={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="card-title "
>
          Luke Skywalker
</h2>,
        <Button
          cardInformation={
                    Object {
                              "Homeworld": "Tatooine",
                              "Homeworld Population": "200000",
                              "Language": "Galactic Basic",
                              "Name": "Luke Skywalker",
                              "Species": "Human",
                              "favorite": false,
                            }
          }
          category={
                    Object {
                              "Homeworld": "Tatooine",
                              "Homeworld Population": "200000",
                              "Language": "Galactic Basic",
                              "Name": "Luke Skywalker",
                              "Species": "Human",
                              "favorite": false,
                            }
          }
          click={[Function]}
          givenClass="favorite-button"
          text=""
/>,
        Array [
          null,
          <div>
            <p
                        className="card-info-title"
            >
                        Species
            </p>
            <p
                        className="card-info-body"
            >
                        Human
            </p>
</div>,
          <div>
            <p
                        className="card-info-title"
            >
                        Language
            </p>
            <p
                        className="card-info-body"
            >
                        Galactic Basic
            </p>
</div>,
          <div>
            <p
                        className="card-info-title"
            >
                        Homeworld Population
            </p>
            <p
                        className="card-info-body"
            >
                        200000
            </p>
</div>,
          <div>
            <p
                        className="card-info-title"
            >
                        Homeworld
            </p>
            <p
                        className="card-info-body"
            >
                        Tatooine
            </p>
</div>,
          null,
        ],
      ],
      "className": "card card-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
          "className": "card-title ",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "cardInformation": Object {
            "Homeworld": "Tatooine",
            "Homeworld Population": "200000",
            "Language": "Galactic Basic",
            "Name": "Luke Skywalker",
            "Species": "Human",
            "favorite": false,
          },
          "category": Object {
            "Homeworld": "Tatooine",
            "Homeworld Population": "200000",
            "Language": "Galactic Basic",
            "Name": "Luke Skywalker",
            "Species": "Human",
            "favorite": false,
          },
          "click": [Function],
          "givenClass": "favorite-button",
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": "Species - Luke Skywalker",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card-info-title"
>
              Species
</p>,
            <p
              className="card-info-body"
>
              Human
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Species",
              "className": "card-info-title",
            },
            "ref": null,
            "rendered": "Species",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Human",
              "className": "card-info-body",
            },
            "ref": null,
            "rendered": "Human",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Language - Luke Skywalker",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card-info-title"
>
              Language
</p>,
            <p
              className="card-info-body"
>
              Galactic Basic
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Language",
              "className": "card-info-title",
            },
            "ref": null,
            "rendered": "Language",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Galactic Basic",
              "className": "card-info-body",
            },
            "ref": null,
            "rendered": "Galactic Basic",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Homeworld Population - Luke Skywalker",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card-info-title"
>
              Homeworld Population
</p>,
            <p
              className="card-info-body"
>
              200000
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Homeworld Population",
              "className": "card-info-title",
            },
            "ref": null,
            "rendered": "Homeworld Population",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "200000",
              "className": "card-info-body",
            },
            "ref": null,
            "rendered": "200000",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Homeworld - Luke Skywalker",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card-info-title"
>
              Homeworld
</p>,
            <p
              className="card-info-body"
>
              Tatooine
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Homeworld",
              "className": "card-info-title",
            },
            "ref": null,
            "rendered": "Homeworld",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Tatooine",
              "className": "card-info-body",
            },
            "ref": null,
            "rendered": "Tatooine",
            "type": "p",
          },
        ],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="card-title "
>
            Luke Skywalker
</h2>,
          <Button
            cardInformation={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Homeworld Population": "200000",
                                    "Language": "Galactic Basic",
                                    "Name": "Luke Skywalker",
                                    "Species": "Human",
                                    "favorite": false,
                                  }
            }
            category={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Homeworld Population": "200000",
                                    "Language": "Galactic Basic",
                                    "Name": "Luke Skywalker",
                                    "Species": "Human",
                                    "favorite": false,
                                  }
            }
            click={[Function]}
            givenClass="favorite-button"
            text=""
/>,
          Array [
            null,
            <div>
              <p
                            className="card-info-title"
              >
                            Species
              </p>
              <p
                            className="card-info-body"
              >
                            Human
              </p>
</div>,
            <div>
              <p
                            className="card-info-title"
              >
                            Language
              </p>
              <p
                            className="card-info-body"
              >
                            Galactic Basic
              </p>
</div>,
            <div>
              <p
                            className="card-info-title"
              >
                            Homeworld Population
              </p>
              <p
                            className="card-info-body"
              >
                            200000
              </p>
</div>,
            <div>
              <p
                            className="card-info-title"
              >
                            Homeworld
              </p>
              <p
                            className="card-info-body"
              >
                            Tatooine
              </p>
</div>,
            null,
          ],
        ],
        "className": "card card-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
            "className": "card-title ",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "cardInformation": Object {
              "Homeworld": "Tatooine",
              "Homeworld Population": "200000",
              "Language": "Galactic Basic",
              "Name": "Luke Skywalker",
              "Species": "Human",
              "favorite": false,
            },
            "category": Object {
              "Homeworld": "Tatooine",
              "Homeworld Population": "200000",
              "Language": "Galactic Basic",
              "Name": "Luke Skywalker",
              "Species": "Human",
              "favorite": false,
            },
            "click": [Function],
            "givenClass": "favorite-button",
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": "Species - Luke Skywalker",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="card-info-title"
>
                Species
</p>,
              <p
                className="card-info-body"
>
                Human
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Species",
                "className": "card-info-title",
              },
              "ref": null,
              "rendered": "Species",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Human",
                "className": "card-info-body",
              },
              "ref": null,
              "rendered": "Human",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Language - Luke Skywalker",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="card-info-title"
>
                Language
</p>,
              <p
                className="card-info-body"
>
                Galactic Basic
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Language",
                "className": "card-info-title",
              },
              "ref": null,
              "rendered": "Language",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Galactic Basic",
                "className": "card-info-body",
              },
              "ref": null,
              "rendered": "Galactic Basic",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Homeworld Population - Luke Skywalker",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="card-info-title"
>
                Homeworld Population
</p>,
              <p
                className="card-info-body"
>
                200000
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Homeworld Population",
                "className": "card-info-title",
              },
              "ref": null,
              "rendered": "Homeworld Population",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "200000",
                "className": "card-info-body",
              },
              "ref": null,
              "rendered": "200000",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Homeworld - Luke Skywalker",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="card-info-title"
>
                Homeworld
</p>,
              <p
                className="card-info-body"
>
                Tatooine
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Homeworld",
                "className": "card-info-title",
              },
              "ref": null,
              "rendered": "Homeworld",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Tatooine",
                "className": "card-info-body",
              },
              "ref": null,
              "rendered": "Tatooine",
              "type": "p",
            },
          ],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
